[PATCH]: allow custom EXTRA_CFLAGS/LDFLAGS/*_SUPPORT on the make cmd line

Signed-off-by: Peter Korsgaard <jacmet@sunsite.dk>
---
 squashfs-tools/Makefile |   26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

Index: squashfs4.1/squashfs-tools/Makefile
===================================================================
--- squashfs4.1.orig/squashfs-tools/Makefile
+++ squashfs4.1/squashfs-tools/Makefile
@@ -87,11 +87,11 @@ MKSQUASHFS_OBJS = mksquashfs.o read_fs.o
 UNSQUASHFS_OBJS = unsquashfs.o unsquash-1.o unsquash-2.o unsquash-3.o \
 	unsquash-4.o swap.o compressor.o
 
-CFLAGS = $(INCLUDEDIR) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
-	-D_GNU_SOURCE -DCOMP_DEFAULT=\"$(COMP_DEFAULT)\"  -O2 -Wall
+CFLAGS = $(EXTRA_CFLAGS) $(INCLUDEDIR) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
+	-D_GNU_SOURCE -DCOMP_DEFAULT=\"$(COMP_DEFAULT)\" -Wall
 
 LIBS =
-ifdef GZIP_SUPPORT
+ifeq ($(GZIP_SUPPORT),1)
 CFLAGS += -DGZIP_SUPPORT
 MKSQUASHFS_OBJS += gzip_wrapper.o
 UNSQUASHFS_OBJS += gzip_wrapper.o
@@ -99,7 +99,7 @@ LIBS += -lz
 COMPRESSORS += gzip
 endif
 
-ifdef LZMA_SUPPORT
+ifeq ($(LZMA_SUPPORT),1)
 LZMA_OBJS = $(LZMA_DIR)/C/Alloc.o $(LZMA_DIR)/C/LzFind.o \
 	$(LZMA_DIR)/C/LzmaDec.o $(LZMA_DIR)/C/LzmaEnc.o $(LZMA_DIR)/C/LzmaLib.o
 INCLUDEDIR += -I$(LZMA_DIR)/C
@@ -109,7 +109,7 @@ UNSQUASHFS_OBJS += lzma_wrapper.o $(LZMA
 COMPRESSORS += lzma
 endif
 
-ifdef XZ_SUPPORT
+ifeq ($(XZ_SUPPORT),1)
 CFLAGS += -DLZMA_SUPPORT
 MKSQUASHFS_OBJS += xz_wrapper.o
 UNSQUASHFS_OBJS += xz_wrapper.o
@@ -117,7 +117,7 @@ LIBS += -llzma
 COMPRESSORS += lzma
 endif
 
-ifdef LZO_SUPPORT
+ifeq ($(LZO_SUPPORT),1)
 CFLAGS += -DLZO_SUPPORT
 ifdef LZO_DIR
 INCLUDEDIR += -I$(LZO_DIR)/include
@@ -129,8 +129,8 @@ LIBS += $(LZO_LIBDIR) -llzo2
 COMPRESSORS += lzo
 endif
 
-ifdef XATTR_SUPPORT
-ifdef XATTR_DEFAULT
+ifeq ($(XATTR_SUPPORT),1)
+ifeq ($(XATTR_DEFAULT),1)
 CFLAGS += -DXATTR_SUPPORT -DXATTR_DEFAULT
 else
 CFLAGS += -DXATTR_SUPPORT
@@ -142,7 +142,7 @@ endif
 #
 # If LZMA_SUPPORT is specified then LZO_DIR must be specified too
 #
-ifdef LZMA_SUPPORT
+ifeq ($(LZMA_SUPPORT),1)
 ifndef LZMA_DIR
 $(error "LZMA_SUPPORT requires LZMA_DIR to be also defined")
 endif
@@ -151,8 +151,8 @@ endif
 #
 # Both XZ_SUPPORT and LZMA_SUPPORT cannot be specified
 #
-ifdef XZ_SUPPORT
-ifdef LZMA_SUPPORT
+ifeq ($(XZ_SUPPORT),1)
+ifeq ($(LZMA_SUPPORT),1)
 $(error "Both XZ_SUPPORT and LZMA_SUPPORT cannot be specified")
 endif
 endif
@@ -175,7 +175,7 @@ endif
 all: mksquashfs unsquashfs
 
 mksquashfs: $(MKSQUASHFS_OBJS)
-	$(CC) $(MKSQUASHFS_OBJS) -lpthread -lm $(LIBS) -o $@
+	$(CC) $(EXTRA_LDFLAGS) $(MKSQUASHFS_OBJS) -lpthread -lm $(LIBS) -o $@
 
 mksquashfs.o: mksquashfs.c squashfs_fs.h mksquashfs.h global.h sort.h \
 	squashfs_swap.h xattr.h
@@ -195,7 +195,7 @@ xattr.o: xattr.h
 read_xattrs.o: xattr.h
 
 unsquashfs: $(UNSQUASHFS_OBJS)
-	$(CC) $(UNSQUASHFS_OBJS) -lpthread -lm $(LIBS) -o $@
+	$(CC) $(EXTRA_LDFLAGS) $(UNSQUASHFS_OBJS) -lpthread -lm $(LIBS) -o $@
 
 unsquashfs.o: unsquashfs.h unsquashfs.c squashfs_fs.h squashfs_swap.h \
 	squashfs_compat.h global.h xattr.h
